go.mod

```diff
@@ -1,13 +1,22 @@
 module github.com/segmentio/kafka-go
 
-go 1.15
+go 1.18
```

Comment:
I think we need a separate PR for upgrading the Go version. Since kafka-go is so widely used this will be a breaking change for some users and the more explicit we can make that the less of a surprise it will be to users